yt_dlp/extractor/tviplayer.py

```diff
@@ -0,0 +1,64 @@
+from .common import InfoExtractor
+from ..utils import traverse_obj
+
+
+class TVIPlayerIE(InfoExtractor):
+    _VALID_URL = r'https?://tviplayer\.iol\.pt(/programa/[\w-]+/[a-f0-9]+)?/video/(?P<id>[a-f0-9]+)'
+    _TESTS = [{
+        'url': 'https://tviplayer.iol.pt/programa/jornal-das-8/53c6b3903004dc006243d0cf/video/61c8e8b90cf2c7ea0f0f71a9',
+        'info_dict': {
+            'id': '61c8e8b90cf2c7ea0f0f71a9',
+            'ext': 'mp4',
+            'duration': 4167,
+            'title': 'Jornal das 8 - 26 de dezembro de 2021',
+            'thumbnail': 'https://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/61c8ee630cf2cc58e7d98d9f/',
+            'season_number': 8,
+            'season': 'Season 8',
+        }
+    }, {
+        'url': 'https://tviplayer.iol.pt/programa/isabel/62b471090cf26256cd2a8594/video/62be445f0cf2ea4f0a5218e5',
+        'info_dict': {
+            'id': '62be445f0cf2ea4f0a5218e5',
+            'ext': 'mp4',
+            'duration': 3255,
+            'season': 'Season 1',
+            'title': 'Isabel - Episódio 1',
+            'thumbnail': 'https://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/62beac200cf2f9a86eab856b/',
+            'season_number': 1,
+        }
+    }, {
+        'url': 'https://tviplayer.iol.pt/video/62c4131c0cf2f9a86eac06bb',
+        'info_dict': {
+            'id': '62c4131c0cf2f9a86eac06bb',
+            'ext': 'mp4',
+            'title': 'David e Mickael Carreira respondem: «Qual é o próximo a ser pai?»',
+            'thumbnail': 'https://www.iol.pt/multimedia/oratvi/multimedia/imagem/id/62c416490cf2ea367d4433fd/',
+            'season': 'Season 2',
+            'duration': 148,
+            'season_number': 2,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        wms_auth_sign_token = self._download_webpage(
+            'https://services.iol.pt/matrix?userId=', video_id,
+            note='Trying to get wmsAuthSign token')
```

Comment:
Shouldn't this be in _real_initialize?

Reply:
Can i save this in self or is there any other method to get this variable if in `_real_initialize` ?

Reply:
Yes, you can save in self

Reply:
The benefit is that if user passes multiple URLs like `yt-dlp URL1 URL2 URL3`, real_initialize will only run once

Reply:
The `_real_initialize` didn't accept pass url by default. How to fix this ?

Reply:
You don't need URL to get the token... What am I missing?

Reply:
Sorry, i forget we can set `video_id` in the `_download_webpage` manually

Reply:
Implemented in https://github.com/yt-dlp/yt-dlp/pull/4281/commits/c0734fc43d867ce7a49a36b2f08fe77805ac0792